test/e2e/knit_irqwatch_test.go

```diff
@@ -0,0 +1,228 @@
+		// select random cpuid to change its counters
+		cpuid := int(rand.Intn(numOfCpus))
+
+		// select random IRQ in the seleced cpuid
+		k := selRandKey(stats[cpuid])
+
+		stats[cpuid][k] += uint64(rand.Intn(maxInterrupts))
+		delta[cpuid] = make(irqs.Counter)
+		delta[cpuid][k] = stats[cpuid][k] - initialStats[cpuid][k]
+	}
+
+	// add the random new stat values to the interruts file
+	return delta, reWriteInterrupts(procFs, stats)
+}
+
+func reWriteInterrupts(procFs string, stats irqs.Stats) error {
+	tmpInterrruptsFile := filepath.Join(procFs, "interrupts_temp")
+	interrruptsFile := filepath.Join(procFs, "interrupts")
+
+	f, err := os.Create(tmpInterrruptsFile)
+	if err != nil {
+		g.Fail(fmt.Sprintf("fail to create temp interrupts file %q", tmpInterrruptsFile))
+		return err
+	}
+
+	defer f.Close()
+
+	// Since it's a test we can assume no offlined cpu, so no holes in the slice (i.e. cpu_0 to cpu_n are presents)
+	type cpuidToIrqValue []uint64
+
+	// IRQ name maps to counter values for each cpuid
+	// this way is makes it easier to be wrriten back into a file, line by line
+	irqMap := make(map[string]cpuidToIrqValue)
+
+	cpuids := getSortedCPUids(stats)
+
+	for cpuid := range cpuids {
+		for k, v := range stats[cpuid] {
+			irqMap[k] = append(irqMap[k], v)
+		}
+	}
+
+	tab(f)
+	// write cpu ids on the first line
+	for cpuid := range cpuids {
+		cpuName := fmt.Sprintf("%11s", fmt.Sprintf("CPU%d", cpuid))
+		f.WriteString(cpuName)
+	}
+	newLine(f)
+
+	for irqName, counters := range irqMap {
+		f.WriteString(fmt.Sprintf("%4s:", irqName))
+
+		for _, v := range counters {
+			f.WriteString(fmt.Sprintf("%11d", v))
+		}
+		newLine(f)
+	}
+
+	// use this approach in order to avoid from other processes to access
+	// the original interrupts file while modifing it
+	os.Rename(tmpInterrruptsFile, interrruptsFile)
+	if err != nil {
+		g.Fail(fmt.Sprintf("fail to rename file %q to %q", tmpInterrruptsFile, interrruptsFile))
+		return err
+	}
+
+	return nil
+}
+
+func tab(f *os.File) {
+	f.WriteString("\t")
+}
+
+func newLine(f *os.File) {
+	f.WriteString("\n")
+}
+
+func selRandKey(c irqs.Counter) string {
+	//fmt.Println(len(c))
+	i := rand.Intn(len(c))
+	for k := range c {
+		if i == 0 {
+			return k
+		}
+	}
+	i--
+	// default key, never expect to get here
+	return "0"
+}
+
+func getSortedCPUids(s irqs.Stats) []int {
+	keys := make([]int, 0, len(s))
+	for k := range s {
+		keys = append(keys, k)
+	}
+	sort.Ints(keys)
+	return keys
+}
+
+func extractCountersFromIRQWatchOutput(b []byte) (irqs.Stats, error) {
+	type irqSummary struct {
+		Elapsed  time.Duration `json:"elapsed"`
+		Counters irqs.Stats    `json:"counters"`
+	}
+
+	irqS := irqSummary{}
+	if err := json.Unmarshal(b, &irqS); err != nil {
+		return nil, fmt.Errorf("Error unmarshalling b: %v", err)
+	}
+
+	return irqS.Counters, nil
+}
+
+func stats2JSON(s irqs.Stats) ([]byte, error) {
+	cpus := cpuset.CPUSet{}
+	cnt := struct {
+		Counters irqs.Stats
+	}{
+		Counters: cm.CountersForCPUs(cpus, s),
```

Comment:
@fromanirh  ```CountersForCPUs``` was taken from [here](https://github.com/fromanirh/debug-tools/blob/irqwatch-cmd/pkg/knit/cmd/irqwatch.go#L226)
Please replace it with a capital 'C'